manifests/plugin/host_reports.pp

```diff
@@ -0,0 +1,58 @@
+# = Foreman Proxy Host Reports plugin
+#
+# This class installs the Host Reports plugin
+#
+# === Parameters:
+#
+# $spool_dir::                Spool directory with processed reports
+#
+# $keep_reports::             Keep sent reports in spool_dir directory
+#                             when enabled, move files from the place on
+#                             a regular basis (e.g. via cronjob).
+#
+# $report_type::              Set to "proxy" to actually enable the new format of reports,
+#                             set to "foreman" for no changes.
+#
+# === Advanced parameters:
+#
+# $proxy_name::               Proxy hostname to appear in reports JSON
+#
+# $enabled::                  enables/disables the host_reports plugin
+#
+# $listen_on::                proxy feature listens on http, https, or both
+#
+# $version::                  plugin package version, it's passed to ensure parameter of package resource
+#                             can be set to specific version number, 'latest', 'present' etc.
+#
+class foreman_proxy::plugin::host_reports (
+  Optional[String] $proxy_name = undef,
+  Stdlib::Absolutepath $spool_dir = '/var/lib/foreman-proxy/host_reports',
+  Boolean $keep_reports = false,
+  # No reports are redirected by default
+  Enum['foreman', 'proxy'] $report_type = 'foreman',
+  Boolean $enabled = true,
+  Foreman_proxy::ListenOn $listen_on = 'https',
+  Optional[String] $version = undef,
+) {
+  $reported_proxy_hostname = pick($proxy_name, $foreman_proxy::registered_name)
+
+  file_line { 'set report type in proxy ansible.cfg':
+    ensure => present,
+    path   => "${foreman_proxy::config_dir}/ansible.cfg",
+    line   => "report_type = ${report_type}",
+    after  => '^[callback_foreman]',
+  }
+  -> file_line { 'set proxy url in proxy ansible.cfg':
+    ensure => present,
+    path   => "${foreman_proxy::config_dir}/ansible.cfg",
+    line   => "proxy_url = ${foreman_proxy::real_registered_proxy_url}",
+    after  => '^[callback_foreman]',
+  }
```

Comment:
You can't do this reliably since the Ansible plugin already manages the file. To have that as leading and modify the template:
https://github.com/theforeman/puppet-foreman_proxy/blob/master/templates/plugin/ansible.cfg.erb

It may mean that you need to introduce additional parameters in `ansible.pp`